src/parser/Script.cpp

```diff
@@ -447,11 +447,48 @@ Value Script::executeModule(ExecutionState& state, Optional<Script*> referrer)
     return resultValue;
 }
 
+// https://www.ecma-international.org/ecma-262/9.0/#sec-candeclareglobalfunction
+static bool canDeclareGlobalFunction(ExecutionState& state, Object* globalObject, AtomicString N)
+{
+    // Let envRec be the global Environment Record for which the method was invoked.
+    // Let ObjRec be envRec.[[ObjectRecord]].
+    // Let globalObject be the binding object for ObjRec.
+    // Let existingProp be ? globalObject.[[GetOwnProperty]](N).
+    auto existingProp = globalObject->getOwnProperty(state, N);
+    // If existingProp is undefined, return ? IsExtensible(globalObject).
+    if (!existingProp.hasValue()) {
+        return globalObject->isExtensible(state);
+    }
+    // If existingProp.[[Configurable]] is true, return true.
+    if (existingProp.isConfigurable()) {
+        return true;
+    }
+    // If IsDataDescriptor(existingProp) is true and existingProp has attribute values { [[Writable]]: true, [[Enumerable]]: true }, return true.
+    if (existingProp.isDataProperty() && existingProp.isWritable() && existingProp.isEnumerable()) {
+        return true;
+    }
+    // Return false.
+    return false;
+}
+
+static void testDecalreGlobalFunctions(ExecutionState& state, InterpretedCodeBlock* topCodeBlock, Object* globalObject)
```

Comment:
Typo. `testDeclareGlobalFunctions` would be correct here.

Reply:
Good catch